Task3/Task3/Task3/Boxes/Box.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Task3.Colors;
+using Task3.Exceptions;
+using Task3.Figures;
+using Task3.FileManagment;
+using Task3.Infrastructure;
+
+namespace Task3.Boxes
+{
+    public class Box
+    {
+        public List<Figure> BoxOfFigures = new List<Figure>(20);
+        
+        private readonly IBoxFileWorker _boxFileWorker;
+
+        public Box()
+        {
+
+        }
+
+        public Box(IBoxFileWorker boxFileWorker)
+        {
+            _boxFileWorker = boxFileWorker;
+        }
+
+        public void AddFigure(Figure item)
+        {
+            var alreadyExist = true;
+
+            foreach (var figure in BoxOfFigures)
+            {
+                if (item.GetType() == figure.GetType() && item.Side == figure.Side)
+                {
+                    alreadyExist = BoxOfFigures.Contains(item);
+                }
+            }
+
+            if (alreadyExist == false)
+            {
+                throw new AddException("This figure is already in the box");
+            }
+            else
+            {
+                BoxOfFigures.Add(item);
+            }
+        }
+
+        public void CheckFigureByNumber(int number)
+        {
+            Console.WriteLine($"{BoxOfFigures[number-1]}, {BoxOfFigures[number-1].Side}");
+        }
+
+        public void RemoveFigure(int number)
+        {
+            BoxOfFigures.Remove(BoxOfFigures[number - 1]);
+            Console.WriteLine(
+                $"Figure {BoxOfFigures[number - 1]}, {BoxOfFigures[number - 1].Side} " +
+                $"with {number} position was deleted");
+        }
+
+        public void ReplaceFigure(Figure item, int position)
+        {
+            BoxOfFigures.Remove(BoxOfFigures[position-1]);
+            BoxOfFigures.Insert(position-1, item);
+        }
+
+        public void CompareFigures(Figure item)
+        {
+            var sameExist = false;
+            var position = 0;
+
+            foreach (var figure in BoxOfFigures)
+            {
+                if (item.GetType() == figure.GetType() && item.Side == figure.Side)
+                {
+                    sameExist = true;
+                    position = BoxOfFigures.IndexOf(figure);
+                }
+            }
+
+            if (sameExist == false)
+            {
+                throw new CompareException("Figure isn't found.");
+            }
+            else
+            {
+                Console.WriteLine(
+                    $"The list has equivalent figure: {BoxOfFigures[position]}, " +
+                    $"with side: {BoxOfFigures[position].Side}");
+            }
```

Comment:
Can we move the action out of the else block?

Reply:
fixed